packages/twenty-server/src/engine/core-modules/workspace/services/workspace-getters.service.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @nx/workspace-inject-workspace-repository */
+import { InjectRepository } from '@nestjs/typeorm';
+import { Injectable, Logger } from '@nestjs/common';
+
+import { Repository } from 'typeorm';
+
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { workspaceValidator } from 'src/engine/core-modules/workspace/workspace.validate';
+import {
+  WorkspaceException,
+  WorkspaceExceptionCode,
+} from 'src/engine/core-modules/workspace/workspace.exception';
+import { isDefined } from 'src/utils/is-defined';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+import { UrlManagerService } from 'src/engine/core-modules/url-manager/service/url-manager.service';
+
+@Injectable()
+export class WorkspaceGettersService {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    private readonly environmentService: EnvironmentService,
+    private readonly urlManagerService: UrlManagerService,
+  ) {}
+
+  async getAuthProvidersByWorkspaceId(workspaceId: string) {
```

Comment:
I would create a module in auth/workspace-auth-provider.module.ts that would only have this

Reply:
I move it into an utils instead of a module.